session.go

```diff
@@ -308,6 +311,14 @@ func (s *session) recvLoop() {
 				continue
 			}
 			c.rb.submit(b)
+
+			if first {
+				if s.ackOnFirst {
+					// immediately send an empty ack to thwart timing attacks
+					c.rb.doSendACK(0)
```

Comment:
The only thing I worry about here is that we're in the receive loop and have experienced those issues with it hanging in the past. Maybe we just throw this on a Go routine since it only happens once/session?

Reply:
Oddly that causes our test to hang. I wouldn't really worry about it since we're sending stuff on this same channel anyway - if we can't send very first thing, we've got bigger problems.

Reply:
OK. TBH it seems like in an ideal world anything we do from this loop that could take time should get shot out to a channel, but I'm cool with this for now.

Reply:
It's already done via a channel - https://github.com/getlantern/lampshade/blob/master/session.go#L638